From db0178fabf2788691dc9144af13d4687966e4f3a Mon Sep 17 00:00:00 2001
From: Peridot <peridot491@protonmail.com>
Date: Fri, 22 Jul 2022 12:55:42 +0200
Subject: [PATCH] Fix PlayerList#sendPacketNearby from Paper-0353 patch

Patch we were using was probably from some fork like Nacho/WindSpigot - now we are using original patch from Paper.
Old code was causing GH-67 (Titanium issue) and probably some other issues.

diff --git a/src/main/java/net/minecraft/server/PlayerList.java b/src/main/java/net/minecraft/server/PlayerList.java
index 7ac4f4a6e..c255ea470 100644
--- a/src/main/java/net/minecraft/server/PlayerList.java
+++ b/src/main/java/net/minecraft/server/PlayerList.java
@@ -1079,12 +1079,12 @@ public abstract class PlayerList {
             // Paper end
 
             // CraftBukkit start - Test if player receiving packet can see the source of the packet
-            if (entityplayer != entityhuman && (world != null || entityplayer.dimension == i)) { // Paper
+            if (entityhuman instanceof EntityPlayer && !entityplayer.getBukkitEntity().canSee(((EntityPlayer) entityhuman).getBukkitEntity())) {
                continue;
             }
             // CraftBukkit end
 
-            if (entityplayer != entityhuman && entityplayer.dimension == i) {
+            if (entityplayer != entityhuman && (world != null || entityplayer.dimension == i)) { // Paper
                 double d4 = d0 - entityplayer.locX;
                 double d5 = d1 - entityplayer.locY;
                 double d6 = d2 - entityplayer.locZ;
-- 
2.36.0.windows.1

